### PR TITLE
Wrap `JS_FORGET_STRING_LINEARNESS`

### PR DESCRIPTION
**Pull request by Redfire75369**

### CODE REVIEW

**Comment by Redfire75369** [comment]
I'm not 100% sure if that `reinterpret_cast` is sound, but it's similar to what I found [here](https://searchfox.org/mozilla-central/source/js/public/Id.h#203), for `PropertyKey::toLinearString`.

**Comment by mrobinson** [comment]
Would it make sense to expose `JSLinearString` as well?

**Comment by Redfire75369** [comment]
`JSLinearString` is already exposed, it's just that it can't be used that much because most functions only take in `JSString`.
Refer to <https://doc.servo.org/servo/index.html?search=JSLinearString> under "In Parameters" and "In Return Types".
I guess it would probably be a good idea to expose `JS_FORGET_STRING_LINEARNESS` (maybe even just that, without this wrapper).

**Comment by mrobinson** [comment]
> `JSLinearString` is already exposed, it's just that it can't be used that much because most functions only take in `JSString`. Refer to https://doc.servo.org/servo/index.html?search=JSLinearString under "In Parameters" and "In Return Types". I guess it would probably be a good idea to expose `JS_FORGET_STRING_LINEARNESS` (maybe even just that, without this wrapper).

That makes sense to me. Thanks!

File: rust-mozjs/src/jsglue.cpp [hunk 1]
```diff
@@ -1018,6 +1018,14 @@ void EncodeStringToUTF8(JSContext* cx, JS::HandleString str,
...
 }
 
+JSString* JS_ForgetStringLinearness(JSLinearString* str) {
+  return JS_FORGET_STRING_LINEARNESS(str);
+}
+
+JSString* EnsureLinearString(JSContext* cx, JSString* str) {
+  return JS_FORGET_STRING_LINEARNESS(JS_EnsureLinearString(cx, str));
+}
+
```
Comment (mrobinson) [inline rust-mozjs/src/jsglue.cpp:1028]: Can this API return a JSLinearString so that it works the same as the wrapped version?

Comment (Redfire75369) [reply rust-mozjs/src/jsglue.cpp:1028]: In that case, this API isn't needed at all. `JS_EnsureLinearString` can already be called from rust.

**Review by mrobinson — APPROVED** [review]
Looks great. Let me know when this is ready to merge.

**Comment by Redfire75369** [comment]
It's ready to merge.

**Comment by Redfire75369** [comment]
It's ready to be merged.

**Pull request merged**